Geospatial.md

```diff
@@ -0,0 +1,144 @@
+* `vincenty`: [https://en.wikipedia.org/wiki/Vincenty%27s_formulae](https://en.wikipedia.org/wiki/Vincenty%27s_formulae)
+* `thomas`: Thomas, Paul D. Spheroidal geodesics, reference systems, & local geometry. US Naval Oceanographic Office, 1970.
+* `andoyer`: Thomas, Paul D. Mathematical models for navigation systems. US Naval Oceanographic Office, 1965.
+* `karney`: [Karney, Charles FF. "Algorithms for geodesics." Journal of Geodesy 87 (2013): 43-55](https://link.springer.com/content/pdf/10.1007/s00190-012-0578-z.pdf), and [GeographicLib](https://geographiclib.sourceforge.io/)
+
+# Logical Types
+
+Two geospatial logical type annotations are supported:
+* `GEOMETRY`: Geometry features in the WKB format with linear/planar edges interpolation. See [Geometry](LogicalTypes.md#geometry)
+* `GEOGRAPHY`: Geography features in the WKB format with an explicit (non-linear/non-planar) edges interpolation algorithm. See [Geography](LogicalTypes.md#geography)
+
+# Statistics
+
+`GeometryStatistics` is a struct specific for `GEOMETRY` and `GEOGRAPHY` logical
+types to store statistics of a column chunk. It is an optional field in the
+`ColumnMetaData` and contains [Bounding Box](#bounding-box) and [Geometry
+Types](#geometry-types) that are described below in detail.
+
+## Bounding Box
+
+A geometry has at least two coordinate dimensions: X and Y for 2D coordinates
+of each point. A geometry can optionally have Z and / or M values associated
+with each point in the geometry.
+
+The Z values introduce the third dimension coordinate. Usually they are used to
+indicate the height, or elevation.
+
+M values are an opportunity for a geometry to express a fourth dimension as a
+coordinate value. These values can be used as a linear reference value (e.g.,
+highway milepost value), a timestamp, or some other value as defined by the CRS.
+
+Bounding box is defined as the thrift struct below in the representation of
+min/max value pair of coordinates from each axis. Note that X and Y Values are
+always present. Z and M are omitted for 2D geometries.
+
+For the X and Y values only, (xmin/ymin) may be greater than (xmax/ymax). In this
+X case, an object in this bounding box may match if it contains an X such that
+`x >= xmin` OR `x <= xmax`, and in this Y case if `y >= ymin` OR `y <= ymax`.
+In geographic terminology, the concepts of `xmin`, `xmax`, `ymin`, and `ymax`
+are also known as `westernmost`, `easternmost`, `southernmost` and `northernmost`,
+respectively.
+
+For `GEOGRAPHY` types, X and Y values are restricted to the canonical ranges of
+[-180, 180] for X and [-90, 90] for Y.
+
+```thrift
+struct BoundingBox {
+  1: required double xmin;
+  2: required double xmax;
+  3: required double ymin;
+  4: required double ymax;
+  5: optional double zmin;
+  6: optional double zmax;
+  7: optional double mmin;
+  8: optional double mmax;
+}
+```
+
+## Geometry Types
+
+A list of geometry types from all geometries in the `GEOMETRY` or `GEOGRAPHY`
```

Comment:
Nit: on same thought (geometry may be misleading , as we representing both), maybe remove `all geometries` , just `A list of geometry types in the 'GEOMETRY' or 'GEOGRAPHY' column'`